### PR TITLE
Don't send playing before prerolls, just after (and major commenting for redispatch)

### PR DESCRIPTION
**Pull request by incompl**

### CODE REVIEW

File: src/videojs.ads.js [hunk 1]
```diff
@@ -337,43 +338,119 @@ var
...
+            player.ads.videoElementRecycled()) {
+
+          // Triggering an event prevents the unprefixed one from firing.
+          // "adcontentplaying" is only seen in this very specific condition.
+          triggerEvent('adcontent', event);
+          return;
+
+        // When ad is playing in content tech, prefix everything with "ad".
+        // This block catches many events such as emptied, play, timeupdate, and ended.
+        } else if (player.ads.state === 'ad-playback') {
```
Comment (ldayananda) [inline src/videojs.ads.js]: I like this change and think this makes the event flow much easier to flow. However, this is likely to affect other plugins relying on these events and so should be thoroughly documented.

Comment (incompl) [inline src/videojs.ads.js]: Yeah I already know of one place where the change to "playing" would break something else. This will be tricky

**Comment by ldayananda** [comment]
LGTM

**Comment by pagarwal123** [comment]
QA: LGTM

**Pull request merged**